biogeochem/EDPhysiologyMod.F90

```diff
@@ -1242,92 +1251,322 @@ subroutine phenology_leafonoff(currentSite)
+             call endrun(msg=errMsg(sourcefile, __LINE__))
+          end if
+
+          if(fates_pft.eq.0)then
+             write(fates_log(),*) 'PFT0 in SP mode'
+             call endrun(msg=errMsg(sourcefile, __LINE__))
+          end if
+
+          ! Call routine to invert SP drivers into cohort properites. 
+          call assign_cohort_SP_properties(currentCohort, currentSite%sp_htop(fates_pft), currentSite%sp_tlai(fates_pft)     , currentSite%sp_tsai(fates_pft),currentPatch%area,ifalse,leaf_c)
+
+          currentCohort => currentCohort%shorter
+       end do !cohort loop
+       currentPatch => currentPatch%younger
+    end do ! patch loop
+
+  end subroutine satellite_phenology
+
+  ! =====================================================================================
+
+  subroutine assign_cohort_SP_properties(currentCohort,htop,tlai,tsai,parea,init,leaf_c)
+
+    ! -----------------------------------------------------------------------------------!
+    ! Takes the daily inputs of leaf area index, stem area index and canopy height and
+    ! translates them into a FATES structure with one patch and one cohort per PFT
+    ! The leaf area of the cohort is modified each day to match that asserted by the HLM
+    ! -----------------------------------------------------------------------------------!
+    use EDTypesMod       , only : nclmax
+
+    type(ed_cohort_type), intent(inout), target :: currentCohort
+
+    real(r8), intent(in) :: tlai ! target leaf area index from SP inputs
+    real(r8), intent(in) :: tsai ! target stem area index from SP inputs
+    real(r8), intent(in) :: htop ! target tree height from SP inputs
+    real(r8), intent(in) :: parea ! patch area for this PFT
+    integer, intent(in)  :: init ! are we in the initialization routine? if so do not set leaf_c
+    real(r8), intent(out) ::  leaf_c        ! leaf carbon estimated to generate target tlai
+
+    real(r8) :: dummy_n       ! set cohort n to a dummy value of 1.0   
+    integer  :: fates_pft     ! fates pft numer for weighting loop
+    real(r8) :: spread        ! dummy value of canopy spread to estimate c_area
+    real(r8) :: check_treelai
+    real(r8) :: canopylai(1:nclmax)
+    real(r8) :: fracerr
+    real(r8) :: oldcarea 
+
+    ! Do some checks 
+    if(associated(currentCohort%shorter))then
+       write(fates_log(),*) 'SP mode has >1 cohort'
+       write(fates_log(),*) "SP mode >1 cohort: PFT",currentCohort%pft, currentCohort%shorter%pft
+       write(fates_log(),*) "SP mode >1 cohort: CL",currentCohort%canopy_layer, currentCohort%shorter%canopy_layer
+       call endrun(msg=errMsg(sourcefile, __LINE__))
+    end if
+
+    !------------------------------------------
+    !  Calculate dbh from input height, and c_area from dbh
+    !------------------------------------------
+    currentCohort%hite = htop
+
+    fates_pft = currentCohort%pft
```

Comment:
this variable is sometimes used sometimes not, I recommend removing this and just using the original variable